File: AtCoder/ABC/B/297/a.rb
b = Array.new
r = Array.new
k = -1
gets.chomp.chars.each_with_index do |c, i|
  b << i % 2 if c == 'B'
  r << i if c == 'R'
  k = i if c == 'K'
end
puts (b.uniq.size == 2 and r[0] < k and k < r[1]) ? "Yes" : "No"

File: AtCoder/ABC/B/297/a.py
s = input()
uniq = []
R = []
K = None
for i in range(len(s)):
    c = s[i]
    if c == 'B': uniq.append(i%2)
    if c == 'R': R.append(i)
    if c == 'K': K = i
print("Yes" if len(set(uniq)) == 2 and R[0] < K and K < R[1] else "No")
